core/src/main/java/net/sourceforge/jnlp/cache/CacheUtil.java

```diff
@@ -813,6 +817,35 @@ public static void waitForResources(ApplicationInstance app, ResourceTracker tra
         }
     }
 
+    private static DownloadServiceListener getDownloadServiceListener(final JNLPClassLoader jnlpClassLoader, final String title, final URL[] undownloaded, final DownloadIndicator indicator) {
+        final EntryPoint entryPoint = jnlpClassLoader.getJNLPFile().getEntryPointDesc();
+        String progressClass = null;
+
+        if (entryPoint instanceof ApplicationDesc) {
+            final ApplicationDesc applicationDesc = (ApplicationDesc) entryPoint;
+            progressClass = applicationDesc.getProgressClass();
+        } else if (entryPoint instanceof AppletDesc) {
+            final AppletDesc appletDesc = (AppletDesc) entryPoint;
+            progressClass = appletDesc.getProgressClass();
+        } else if (entryPoint instanceof InstallerDesc) {
+            final InstallerDesc installerDesc = (InstallerDesc) entryPoint;
+            progressClass = installerDesc.getProgressClass();
+        }
+
+        if (progressClass != null) {
+            try {
+                final Class<?> downloadProgressIndicatorClass = jnlpClassLoader.loadClass(progressClass);
+                return (DownloadServiceListener) downloadProgressIndicatorClass.newInstance();
+            } catch (ClassNotFoundException | IllegalAccessException | InstantiationException ex) {
+                LOG.warn("Could not load progress class '{}' specified in JNLP file, " +
```

Comment:
we should add the exception to see the stack trace?

Reply:
fixed